### ISSUE TITLE
The holidays are one day earlier when a "positive" time zone (eg: +01:00) is chosen

### ISSUE THREAD

**Issue by dreamhunt** [Zren/plasma-applet-eventcalendar#15]
Hello, firstly thank you for your great Calendar widget :) 
I found one unpleasant bug - when Events->Holidays is checked the events are shown but are actually one day earlier in Bulgaria. Let say that 24 may is our Culture and Literacy Day but in the widget it is shown one day earlier - 23 may. All holidays are affected by this bug. It'll be great if you can fix it. Thank you :)
P.S.:
When Google Calendar->Holidays in Bulgaria is checked the holidays are shown correctly but are in english. Events->Holidays shows them in Bulgarian language which makes this option prefered. but it's buged for now :)

**Comment by Zren** [comment]
Does it show on the correct date in the Digital Clock widget?

**Comment by dreamhunt** [comment]
Thank you for your replay! 
The holidays doesn't show at all in the Digital Clock.

**Comment by Zren** [comment]
Well it appears to be correct in KHolidays:

https://github.com/KDE/kholidays/blob/master/holidays/plan2/holiday_bg_bg#L26

It's also displayed on the correct date for me as well in Canada.

![](https://i.imgur.com/zv5yp9F.png)

Even testing the bulgarian language.

![](https://i.imgur.com/PwPU7Iq.png)

I don't really want to have to change my timezone... but I might need to try that next.

**Comment by Zren** [comment]
What timezone are you in?

**Comment by dreamhunt** [comment]
Hello, sory for the delay I have to go out for a while. We test it on my Arch installation and on Arch installation of a friend. The bug occurs on both of the machines. My timezone is Europe/Sofia:
![screenshot_20180509_010636](https://user-images.githubusercontent.com/9913684/39785560-3104c926-530c-11e8-9eae-eb2a96f1445a.png)

**Comment by Zren** [comment]
Ah yep, I can reproduce with `TZ="Europe/Sofia" plasmoidviewer ...`

![](https://i.imgur.com/tedGTQW.png)

I'll try to figure out why this is. I probably screwed up converting something to UTC or something.

In the meantime, you can enjoy watching this if you're bored: https://www.youtube.com/watch?v=-5wpm-gesOY

**Comment by dreamhunt** [comment]
Thank you it'll be great! Cool video :)

**Comment by Zren** [comment]
I confirmed DigitalClock displays the dates on May 24 with

`TZ="Europe/Sofia" plasmoidviewer -a org.kde.plasma.digitalclock`

![](https://i.imgur.com/iBi0Kef.png)

So I am parsing the data incorrectly somehow.

I noticed that [the dates grabbed](https://github.com/Zren/plasma-applet-eventcalendar/blob/f856c881dffac9ca5204a9add6c2833974a3ad81/package/contents/ui/calendars/PlasmaCalendarManager.qml#L161) from `calendarBackend.daysModel.eventsForDate(day)

```
qml: [eventcalendar:debug] "2018-05-23T21:00:00.000Z" [
	{
		"objectName": "",
		"startDateTime": "2018-05-23T21:00:00.000Z",
		"endDateTime": "2018-05-23T21:00:00.000Z",
		"isAllDay": true,
		"isMinor": false,
		"title": "Ден на българската просвета и култура и на славянската писменост",
		"description": "",
		"eventColor": "",
		"eventType": "Holidays"
	}
]
qml: [eventcalendar:debug] "2018-05-24T21:00:00.000Z" []
```

Hmmm. Nothing funny with my day iteration code. It preserves the timezone.

![](https://i.imgur.com/IU3TNZb.png)

* Hmmm. If I try `TZ='Asia/Tokyo'`, which is `+09:00`, it also shows the date on May 23.
* `Etc/UTC` aka `+00:00` shows on May 24.
* `Etc/GMT-1` aka `+01:00` shows on May 23.
* and lastly `Etc/GMT+1` aka `-01:00` shows on May 24.

So it seems *all* timezones after GMT are showing the event 1 day early. This means it's probably comparing a the timezone specific date against a UTC date... [here probably](https://github.com/KDE/plasma-framework/blame/master/src/declarativeimports/calendar/daysmodel.cpp#L190) since it's a `QMultiHash<QDate, CalendarEvents::EventData> m_eventsData;`.

So I'll need to strip the timezone from the date somehow...

Well I can use `dateMinUtc = new Date("2018-05-24T00:00:00.000Z")` to make sure it returns events from the right date.

```
qml: [eventcalendar:debug] "2018-05-24T00:00:00.000Z" [
	{
		"objectName": "",
		"startDateTime": "2018-05-23T21:00:00.000Z",
		"endDateTime": "2018-05-23T21:00:00.000Z",
		"isAllDay": true,
		"isMinor": false,
		"title": "Ден на българската просвета и култура и на славянската писменост",
		"description": "",
		"eventColor": "",
		"eventType": "Holidays"
	}
]
```

This fixes "fetching" the events. Now we need to display them on the right date. The date it returns is apprarently UTC, which is parsed as "local time" thus it's printed as 9PM the day before.

I'll tackle it again in a bit.
